File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        n = len(nums)
        max1 = l = 0
        
        for r in range(n):
            if nums[r] == 0:
                l = r + 1
            else:
                max1 = max(max1, r - l + 1)
        return max1

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int max1 = 0;  
        int l = 0;    

        for (int r = 0; r < n; ++r) { 
            if (nums[r] == 0) {
                l = r + 1;
            } else {
                max1 = max(max1, r - l + 1);
            }
        }

        return max1;
    }
};